### PR TITLE
Refactoring of report generator

### PR DESCRIPTION
**Pull request by OnlyDeniko**

### CODE REVIEW

File: report_generator/default_report_gen_config.json [hunk 1]
```diff
@@ -10,9 +10,22 @@
...
+        "log_loss": "2 / 1",
+        "roc_auc": "2 / 1",
+        "rmse": "2 / 1",
+        "r2_score": "2 / 1",
+        "n_sv": "2 / 1",
+        "n_clusters": "2 / 1"
+    },
+    "aggregation_metrics": [
+        "geomean",
+        "average"
```
Comment (PetrovKP) [inline report_generator/default_report_gen_config.json:29]: I think default better without average

File: report_generator/default_report_gen_config.json [hunk 2]
```diff
@@ -10,9 +10,21 @@
...
+        "inertia": "2 / 1",
+        "iter": "2 / 1",
+        "noise_variance": "2 / 1",
+        "accuracy": "2 / 1",
+        "log_loss": "2 / 1",
+        "roc_auc": "2 / 1",
+        "rmse": "2 / 1",
+        "r2_score": "2 / 1",
+        "n_sv": "2 / 1",
+        "n_clusters": "2 / 1"
```
Comment (Alexsandruss) [inline report_generator/default_report_gen_config.json:25]: Too many same values `"2 / 1"`. What about adding `"default": "2 / 1"`?

File: report_generator/report_generator.py [hunk 3]
```diff
@@ -18,82 +18,189 @@
...
+
+def write_header_of_sheet(
+    work_sheet,
+    header_columns: List[str],
+    y_offset: int,
+    metrics: List[str],
+    agg_offset: int,
+    agg_metrics: List[str],
+    json_results: List[Dict[str, Any]],
+    LEFT_OFFSET: int,
```
Comment (Alexsandruss) [inline report_generator/report_generator.py:135]: Switch capital to lower case in function parameter

File: report_generator/report_generator.py [hunk 4]
```diff
@@ -18,82 +18,189 @@
...
+                )
+                metric_offset += 1
+
+
+def get_color_rule(metric: str) -> Any:
+    if metric in ['geomean', 'time[s]']:
+        return ColorScaleRule(
+            start_type='num', start_value=0.5, start_color='FF0000',
+            mid_type='num', mid_value=1, mid_color='FFFF00',
+            end_type='num', end_value=5, end_color='00FF00')
```
Comment (Alexsandruss) [inline report_generator/report_generator.py:187]: Minor: milder colors than `FF` and `00` are better for view.

File: report_generator/report_generator.py [hunk 5]
```diff
@@ -18,82 +18,189 @@
...
+            mid_type='num', mid_value=0, mid_color='FFFF00',
+            end_type='num', end_value=3, end_color='00FF00')
+    return ColorScaleRule(
+        start_type='percentile', start_value=10, start_color='FF0000',
+        mid_type='percentile', mid_value=50, mid_color='FFFF00',
+        end_type='percentile', end_value=90, end_color='00FF00')
+
+
+def get_ratio_string(a: str, b: str, comparison_method: str) -> str:
+    splited_comparison_method = comparison_method.split(' ')
```
Comment (Alexsandruss) [inline report_generator/report_generator.py:200]: Minor: rename `splited` to `splitted` or `split`

**Review by Alexsandruss — CHANGES_REQUESTED** [review]
Remove empty columns like `input_data:classes` in `kmeans` sheet.

File: report_generator/default_report_gen_config.json (hunk 2 above)
Comment (OnlyDeniko) [reply report_generator/default_report_gen_config.json:25]: Good idea

**Comment by OnlyDeniko** [comment]
> Remove empty columns like `input_data:classes` in `kmeans` sheet.

Really good idea, thank you. Now if report does not have some parameters from config['header'] then they will not be shown.

File: report_generator/README.md [hunk 6]
```diff
@@ -2,10 +2,14 @@
...
 runner options:
 * ``result-files`` : comma-separated benchmark json result file paths
 * ``report-file`` : report file path
 * ``generation-config`` : generation configuration file path
-* ``merging``: *full*, *none*, *sw_only*, *hw_only*. How to merge same cases in benchmark logs
+
+config parameters:
+* ``header`` : Column names in the table header. These parameters are also used to compare reports. If the name is compound, then it should be separated by the '':'' symbol.
+* ``comparison_method`` : Comparison of two results. The options are: ``1 operation 2`` or ``2 operation 1``. Default is ``2 / 1``.
+* ``aggregation_metrics`` : Metric applied to columns with comparisons of two reports. For each of these metrics, a separate sheet with a summary is compiled. It is important that the function is in Excel. For example: ``geomean`` or ``average``.
```
Comment (outoftardis) [inline report_generator/README.md:15]:
```suggestion
* ``aggregation_metrics``: The metrics applied to the columns with the comparisons of two reports. For each of these metrics, a separate sheet with a summary is compiled. Use Excel function names to provide metrics, for example: ``geomean`` or ``average``.
```

If users want to provide multiple metrics, they should provide them as a list, I assume? I think you'd benefit from having examples of how the config file could look like.

File: report_generator/README.md [hunk 7]
```diff
@@ -2,10 +2,14 @@
...
 
 runner options:
 * ``result-files`` : comma-separated benchmark json result file paths
 * ``report-file`` : report file path
 * ``generation-config`` : generation configuration file path
-* ``merging``: *full*, *none*, *sw_only*, *hw_only*. How to merge same cases in benchmark logs
+
+config parameters:
+* ``header`` : Column names in the table header. These parameters are also used to compare reports. If the name is compound, then it should be separated by the '':'' symbol.
+* ``comparison_method`` : Comparison of two results. The options are: ``1 operation 2`` or ``2 operation 1``. Default is ``2 / 1``.
```
Comment (outoftardis) [inline report_generator/README.md:14]:
```suggestion
* ``comparison_method``: The formula for the comparison of two results. The options are: ``1 operation 2`` or ``2 operation 1``, where ``1`` is the first result and ``2`` is the second result. The default is ``2 / 1``, which returns the ratio of the second result to the first one.
```

File: report_generator/README.md [hunk 8]
```diff
@@ -2,10 +2,14 @@
...
+Run `python report_generator.py --result-files bench_log_1.json,bench_log_2.json [--report-file new_report.xlsx --generation-config default_report_gen_config.json]` to launch report generation.
 
 runner options:
 * ``result-files`` : comma-separated benchmark json result file paths
 * ``report-file`` : report file path
 * ``generation-config`` : generation configuration file path
-* ``merging``: *full*, *none*, *sw_only*, *hw_only*. How to merge same cases in benchmark logs
+
+config parameters:
+* ``header`` : Column names in the table header. These parameters are also used to compare reports. If the name is compound, then it should be separated by the '':'' symbol.
```
Comment (outoftardis) [inline report_generator/README.md:13]:
```suggestion
* ``header``: The column names in the table header. These parameters are also used to compare reports. If a name is compound, use the ``:`` symbol to separate its parts.
```

File: report_generator/README.md (hunk 6 above)
Comment (OnlyDeniko) [reply report_generator/README.md:15]:
```
"aggregation_metrics": [
        "geomean",
        "average"
    ]
```

**Review by Alexsandruss — APPROVED** [review]

File: report_generator/README.md [hunk 9]
```diff
@@ -2,10 +2,14 @@
...
 runner options:
 * ``result-files`` : comma-separated benchmark json result file paths
 * ``report-file`` : report file path
 * ``generation-config`` : generation configuration file path
-* ``merging``: *full*, *none*, *sw_only*, *hw_only*. How to merge same cases in benchmark logs
+
+config parameters:
+* ``header``: The column names in the table header. These parameters are also used to compare reports. If a name is compound, use the ``:`` symbol to separate its parts.
+* ``comparison_method``: The formula for the comparison of two results. The options are: ``1 operation 2`` or ``2 operation 1``, where ``1`` is the first result and ``2`` is the second result. The default is ``2 / 1``, which returns the ratio of the second result to the first one.
+* ``aggregation_metrics`` : Metrics applied to columns with comparisons of two reports.  You can use multiple metrics. For each of these metrics, a separate sheet with a summary is compiled. It is important that the function is in Excel. For example: ``"geomean", "average"``.
```
Comment (outoftardis) [inline report_generator/README.md:15]:
```suggestion
* ``aggregation_metrics``: The metrics applied to the columns with the comparisons of two reports. You can use multiple metrics. For each of these metrics, a separate sheet with a summary is compiled. The metrics should be Excel functions. For example: ``"geomean", "average"``.
```

**Review by outoftardis — APPROVED** [review]

**Review by RukhovichIV — APPROVED** [review]

**Pull request merged**